pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -1804,8 +1805,20 @@ in
   };
 
   orgmode = super.orgmode.overrideAttrs {
-    dependencies = with self; [ (nvim-treesitter.withPlugins (p: [ p.org ])) ];
+    buildInputs = [
+      git
+    ];
+    dependencies = with self; [
+      nvim-treesitter-parsers.org
+      # tree-sitter-grammars.tree-sitter-org-nvim
+      plenary-nvim
+    ];
     nvimRequireCheck = "orgmode";
+    postInstall = ''
+      mkdir -p tests/.deps/plugins
+      cp -r ${self.plenary-nvim} tests/.deps/plugins/plenary
+    '';
+    nvimTestCheck = "tests/test.lua";
```

Comment:
this level of abstraction is hard to implement IMO. Plugins are still standarding their test runners. You can have a look at a few testssuites I run in the overrides.nix (lua or vim, dont remember) and they all need some tweaking.
Hopefully in the future, plugins will converge and `make test` or `busted` or `luarocks test` would be enough.